warehouse/query-core/src/main/java/datawave/query/attributes/AttributeFactory.java

```diff
@@ -56,17 +55,15 @@ public AttributeFactory(TypeMetadata typeMetadata, String defaultType) {
     }
 
     private String extractIngestDataTypeFromKey(Key key) {
-        Text cf = new Text();
-        key.getColumnFamily(cf);
-        ByteBuffer b = ByteBuffer.wrap(cf.getBytes(), 0, cf.getLength());
-        int endPos = 0;
-        while (b.hasRemaining()) {
-            if (b.get() == 0)
+        ByteSequence bytes = key.getColumnFamilyData();
+        int index = 0;
+        for (int i = 0; i < bytes.length(); i++) {
+            if (bytes.byteAt(i) == 0x00) {
+                index = i;
                 break;
-            endPos++;
+            }
         }
-        cf.set(cf.getBytes(), 0, endPos);
-        return cf.toString();
+        return bytes.subSequence(0, index).toString();
```

Comment:
I see that `ByteSequence` in the 2.1 or main Accumulo branch doesn't seem to implement `toString()`, does this mean that this should really something like:
```
return new String(bs.subSequence(0, index).toArray(), StandardCharsets.UTF_8)
```

Reply:
Thanks for pointing this out - this is why looking at code in GitHub doesn't always work the best.